src/Microsoft.ML.TensorFlow/TensorTypeExtensions.cs

```diff
@@ -0,0 +1,121 @@
+﻿using System;
+using NumSharp.Backends;
+using NumSharp.Backends.Unmanaged;
+using NumSharp.Utilities;
+using Tensorflow;
+
+namespace Microsoft.ML.Transforms
+{
+    [BestFriend]
+    internal static class TensorTypeExtensions
+    {
+        public static void ToArray<T>(this Tensor tensor, T[] array) where T: unmanaged
+        {
+            var size = array.Length;
+            if (typeof(T).as_dtype() == tensor.dtype)
+            {
+                if (tensor.NDims == 0 && size == 1)
+                {
+                    unsafe
+                    {
+                        array[0] = *(T*)tensor.buffer;
+                    }
+                }
+
+                unsafe
+                {
+                    var len = (long)size;
+                    fixed (T* dst = array)
+                    {
+                        var src = (T*)tensor.buffer;
+                        len *= ((long)tensor.itemsize);
+                        System.Buffer.MemoryCopy(src, dst, len, len);
+                    }
+                }
+            }
+            else
+            {
+                if (tensor.NDims == 0 && size == 1) //is it a scalar?
+                {
+                    unsafe
+                    {
+#if _REGEN
```

Comment:
I suggest to add a comment mentioning that this is a template written in Regen with a url:
https://github.com/Nucs/Regen

Reply:
Nevermind, I just realized it was removed later on #resolved